crypto-test/src/test/scala/org/bitcoins/crypto/BitcoinSCryptoTest.scala

```diff
@@ -0,0 +1,118 @@
+package org.bitcoins.crypto
+
+import org.scalacheck.Gen
+import org.scalactic.anyvals.PosInt
+import org.scalatest.flatspec.{AnyFlatSpec, AsyncFlatSpec}
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.{Assertion, BeforeAndAfter, BeforeAndAfterAll, Succeeded}
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait BitcoinSCryptoTest
+    extends AnyFlatSpec
+    with BeforeAndAfter
+    with BeforeAndAfterAll
+    with Matchers
+    with ScalaCheckPropertyChecks {
+
+  implicit def executionContext: ExecutionContext =
+    scala.concurrent.ExecutionContext.global
+
+  def generatorDrivenConfigNewCode: PropertyCheckConfiguration = {
+    customGenDrivenConfig(BitcoinSCryptoTest.NEW_CODE_EXECUTIONS)
+  }
+
+  /** Sets the generator driven tests to perform the given amount of execs */
+  def customGenDrivenConfig(executions: Int): PropertyCheckConfiguration = {
+    PropertyCheckConfiguration(
+      minSuccessful = PosInt.from(executions).get,
+      minSize = PosInt.from(executions).get,
+      workers = 1
+    )
+  }
+
+}
+
+trait BitcoinSCryptoAsyncTest
+    extends AsyncFlatSpec
+    with BeforeAndAfter
+    with BeforeAndAfterAll
+    with Matchers
+    with ScalaCheckPropertyChecks {
+
+  implicit override def executionContext =
+    scala.concurrent.ExecutionContext.Implicits.global
+
+  def generatorDrivenConfigNewCode: PropertyCheckConfiguration = {
+    customGenDrivenConfig(BitcoinSCryptoTest.NEW_CODE_EXECUTIONS)
+  }
+
+  /** Sets the generator driven tests to perform the given amount of execs */
+  def customGenDrivenConfig(executions: Int): PropertyCheckConfiguration = {
+    PropertyCheckConfiguration(
+      minSuccessful = PosInt.from(executions).get,
+      minSize = PosInt.from(executions).get,
+      workers = 1
+    )
+  }
+
+  def forAllAsync[A](gen: Gen[A])(
```

Comment:
Ok, so just for my own knowledge, was the problem with the previous implemenation was the fact that it used `java.util.concurrent.CopyOnwriteList` ? 

https://github.com/bitcoin-s/bitcoin-s/blob/ac78b60ea0e3e659a0729646074daba0f6f31eb5/testkit-core/src/main/scala/org/bitcoins/testkitcore/util/BaseAsyncTest.scala#L94

If so, i vote that we get rid of the old implementation and replace it with this one. Your implementation seems superior